src/psc_datalogger/connection.py

```diff
@@ -492,3 +509,41 @@ def _exit(self) -> None:
         # Note it will do 1 more iteration of the loop, inlcuding the sleep
         self.running = False
         self.logging_signal.set()
+
+    def _connection_check(self) -> None:
+        """Check whether connection is initialized.
+
+        Mostly required to guard against potential checks before the Prologix
+        controller has connected (or maybe there isn't even one plugged in yet!)"""
+        if not hasattr(self, "connection"):
+            logging.warning(
+                "Attempted to write to connection when it was not initialized"
+            )
+            raise ConnectionNotInitialized()
+
+    def _connection_write(self, command: str) -> None:
+        """Wrapper for all connection write commands."""
```

Comment:
The fact that it is a wrapper isn't that relevant as a description. You're checking the connection before attempting a write operation.

Also, the fact that these methods are all prepended with 'connection' suggests that this abstraction would make sense as a separate class etc, but this program is small enough that it isn't too important.

Reply:
Updated in 12288e0aab432a492f497cddbf445f9c2dd1fce5